### PR TITLE
Update to newest otel libs and rust 1.76

### PR DESCRIPTION
**Pull request by Licenser**
A bunch of updates for the newest otel and rust 1.76 ideomatic code main changes are:

* Update tracing to newest libs
* use and_then to avoide needing to implement chaining manually on layers
* use `match!` in test to avoid PartialEq on errors
* update enviroment variable handling to allow overwriting URL
* include `/api` in default url to allow local testing
* include shutdown in examples so we don't have to hardcode it
* tidy cargo.toml
* move examples to be runnable via `cargo run --example ...`
* change error env var error type to `'static str` to avoide need for allocations

### CODE REVIEW

**Review by darach — APPROVED** [review]
Like! Very nice set of udpates 🎈

**Review by darach — APPROVED** [review]
Like! Very nice set of udpates 🎈

File: src/builder.rs [hunk 1]
```diff
@@ -1,57 +1,42 @@
...
+    registry::LookupSpan,
+    util::SubscriberInitExt,
+    Layer, Registry,
 };
 
-use crate::Error;
-use reqwest::Url;
-
-const CLOUD_URL: &str = "https://cloud.axiom.co";
+const CLOUD_URL: &str = "https://cloud.axiom.co/api";
```
Comment (Licenser) [inline src/builder.rs]:
```suggestion
const CLOUD_URL: &str = "https://api.axiom.co";
```

File: src/builder.rs [hunk 2]
```diff
@@ -345,30 +301,27 @@ mod tests {
...
 
     #[tokio::test(flavor = "multi_thread")]
     async fn test_no_env_skips_env_variables() -> Result<(), Error> {
         let builder = Builder::new().no_env();
-        assert_eq!(builder.no_env, true);
+        assert!(builder.no_env);
         assert_eq!(builder.token, None);
         assert_eq!(builder.dataset_name, None);
-        assert_eq!(builder.url, Some("https://cloud.axiom.co".into()));
+        assert_eq!(builder.url, Some("https://cloud.axiom.co/api".into()));
```
Comment (Licenser) [inline src/builder.rs]:
```suggestion
        assert_eq!(builder.url, Some("https://api.axiom.co".into()));
```

File: examples/noenv/main.rs [hunk 3]
```diff
@@ -10,15 +10,18 @@ fn say_hi(id: Uuid, name: impl Into<String> + std::fmt::Debug) {
 async fn main() -> Result<(), Box<dyn std::error::Error>> {
     tracing_axiom::builder()
         .with_tags(&[("aws_region", "us-east-1")]) // Set otel tags
-        .with_dataset("tracing-axiom-examples".to_string()) // Set dataset
+        .with_dataset("tracing-axiom-examples") // Set dataset
         .with_token("xaat-some-valid-token") // Set API token
-        .with_url("Some valid URL other than default") // Set URL
+        .with_url("http://localhost:4318") // Set URL, local Jaeger can be changed to any OTEL endpoint
```
Comment (Licenser) [inline examples/noenv/main.rs]:
```suggestion
        .with_url("http://localhost:4318") // Set URL, can be changed to any OTEL endpoint
```

**Review by bahlo — APPROVED** [review]
nice

**Pull request merged**